spark/src/main/java/org/apache/iceberg/spark/data/SparkOrcValueReaders.java

```diff
@@ -166,12 +169,39 @@ protected void set(InternalRow struct, int pos, Object value) {
 
   static class UnionReader implements OrcValueReader<Object> {
     private final OrcValueReader[] readers;
+    private final Type expected;
```

Comment:
'expected' is derived from the icebergSchema. Can we rename it to 'expectedIcebergSchema'.

Reply:
Good point, done